backend/src/controllers/users/create-user.js

```diff
@@ -0,0 +1,43 @@
+import bcrypt from 'bcrypt';
+
+import { connection as knex } from '../../database/connection.js';
+
+const createUser = async (req, res) => {
+    const {name, email, password} = req.body;
+
+    try {
+        const verifyEmail = await knex('users').select('email').where({email});
+
+        console.log('VERIFICAÇÃO DO EMAIL', verifyEmail);
+
+        if (verifyEmail.length <= 0) {
+            const encryptedPassword = await bcrypt.hash(password, 10);
+
+            const create = await knex('users').insert(
+                {
+                    name,
+                    email,
+                    password: encryptedPassword
+                }
+            ).returning(
+                [
+                    'id',
+                    'name',
+                    'email'
+                ]
+            );
+
+            return res.status(201).json(create);
+        }
+
+        return res.status(400).json({message: 'This email address is already registered.'});
```

Comment:
Acho que em termos de funcionalidade e padronização de código, este último return deveria estar dentro da função validadora, pois se trata de uma exceção e não do que este controlador faz no geral. 
Aí teríamos if (verifyEmail.length > 0)
{return res.status(400).json({message: 'This email address is already registered.'});}
Aí o que estava dentro dessa função anteriormente ficaria fora dela, se tornando o conteúdo principal do controlador.

Reply:
feito